packages/cadl-ranch-specs/cadl-ranch-summary.md

```diff
@@ -1578,6 +1578,189 @@ Scenario that returns a different file encoding depending on the accept header.
 - image/png return a png image
 - image/jpeg return a jpeg image
 
+### Payload_JsonMergePatch_createOptionalResource
+
+- Endpoint: `put /json-merge-patch/create/resource/optional`
+
+Expected input body:
+
+```json
+{
+  "name": "Madge",
+  "description": "desc",
+  "map": {
+    "key": {
+      "name": "InnerMadge",
+      "description": "innerDesc"
+    }
+  },
+  "array": [
+    {
+      "name": "InnerMadge",
+      "description": "innerDesc"
+    }
+  ],
+  "intValue": 1,
+  "floatValue": 1.1
+}
+```
+
+Expected response body:
+
+```json
+{
+  "name": "Madge",
+  "description": "desc",
+  "map": {
+    "key": {
+      "name": "InnerMadge",
+      "description": "innerDesc"
+    }
+  },
+  "array": [
+    {
+      "name": "InnerMadge",
+      "description": "innerDesc"
+    }
+  ],
+  "intValue": 1,
+  "floatValue": 1.1
+}
+```
+
+### Payload_JsonMergePatch_createResource
+
+- Endpoint: `put /json-merge-patch/create/resource`
+
+Expected input body:
+
+```json
+{
+  "name": "Madge",
+  "description": "desc",
+  "map": {
+    "key": {
+      "name": "InnerMadge",
+      "description": "innerDesc"
+    }
+  },
+  "array": [
+    {
+      "name": "InnerMadge",
+      "description": "innerDesc"
+    }
+  ],
+  "intValue": 1,
+  "floatValue": 1.1
+}
+```
+
+Expected response body:
+
+```json
+{
+  "name": "Madge",
+  "description": "desc",
+  "map": {
+    "key": {
+      "name": "InnerMadge",
+      "description": "innerDesc"
+    }
+  },
+  "array": [
+    {
+      "name": "InnerMadge",
+      "description": "innerDesc"
+    }
+  ],
+  "intValue": 1,
+  "floatValue": 1.1
+}
+```
+
+### Payload_JsonMergePatch_updateOptionalResource
+
+- Endpoint: `patch /json-merge-patch/update/resource/optional`
+
+Should serialize null values with merge-patch+json enabled.
+
+Expected input body:
+
+```json
+{
+  "name": "Madge",
+  "description": null,
+  "map": {
+    "key": {
+      "name": "InnerMadge",
+      "description": null
+    }
+  },
+  "array": null,
+  "intValue": null,
+  "floatValue": null
+}
+```
+
+Expected response body:
+
+```json
+{
+  "name": "Madge",
+  "description": null,
+  "map": {
+    "key": {
+      "name": "InnerMadge",
+      "description": null
+    }
+  },
+  "array": null,
+  "intValue": null,
+  "floatValue": null
```

Comment:
Typical response would just having these optional property (of no value) not present.

Reply:
I changed response to `NoContentResponse`

Reply:
`NoContentResponse` may not be a realistic response... I assume typical PATCH would return a resource.

Let's wait for opinion from other reviewers before further change.

Reply:
Updated. Make response as `Resource` and remove the `null` fields in response now.